module/solutions/helper.js

```diff
@@ -1470,6 +1519,173 @@ module.exports = class SolutionsHelper {
+
+        if(!appDetails || !appDetails.data){
+          throw new Error(constants.apiResponses.APP_DETAILS_NOT_FOUND);
+        }
+
+        let link;
+
+        if(!solutionData[0].link){
+
+          let updateLink = await gen.utils.md5Hash(solutionData[0]._id + "###" + solutionData[0].author);
+
+          let updateSolution = await this.updateSolutionDocument
+                (
+                    { _id : solutionId },
+                    { $set : { link : updateLink } }
+                );
+
+          link = await gen.utils.generateLink(appsPortalBaseUrl, appName, updateLink, solutionData[0].type);
+
+        } else {
+          link = await gen.utils.generateLink(appsPortalBaseUrl, appName, solutionData[0].link, solutionData[0].type);
+        }
+
+        return resolve({
+            success : true,
+            message: constants.apiResponses.LINK_GENERATED,
+            result: link
+        });
+
+      } catch(error) {
+        return resolve({
+          success : false,
+          status : error.status ? 
+          error.status : httpStatusCode['internal_server_error'].status,
+          message : error.message
+        })
+      }
+    })
+   }
+
+  /**
+   * Get Details By solution link
+   * @method
+   * @name getDetailsByLink
+   * @param {String} link - link.
+   * @param {String} requestingUserAuthToken - Requesting user auth token.
+   * @param {Object} bodyData - request body data.
+   * @returns {Object} data.
+   */
+
+   static getDetailsByLink( link = "", requestingUserAuthToken = "", userId = "", bodyData = {} ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+
+        if (link == "") {
+            throw new Error(constants.apiResponses.LINK_REQUIRED_CHECK)
+        }
+
+        if (requestingUserAuthToken == "") {
+            throw new Error(constants.apiResponses.REQUIRED_USER_AUTH_TOKEN)
+        }
+
+        if (userId == "") {
+            throw new Error(constants.apiResponses.USER_ID_REQUIRED_CHECK)
+        }
+
+        let solutionData = await this.solutionDocuments({
+                    link: link,
+                    isReusable : false,
+                    status: { $ne : constants.common.IN_ACTIVE }
+                },[
+                    "type","status", "endDate"
+                ]);
+
+        if(!Array.isArray(solutionData) || solutionData.length < 1){
+            return resolve({
+                message: constants.apiResponses.INVALID_LINK,
+                result: []
+            });  
+        }
+
+        if(solutionData[0].status != constants.common.ACTIVE) {
+            return resolve({
+                message: constants.apiResponses.LINK_IS_EXPIRED,
+                result: []
+            });   
+        }
+
+        if (solutionData[0].endDate && new Date() > new Date(solutionData[0].endDate)) {
+            if (solutionData[0].status == constants.common.ACTIVE) {
+                await this.updateSolutionDocument
```

Comment:
@priyanka-TL  Lets use update helper function here also please :mask:

Reply:
@aman-tunerlabs we don't have a helper function for update solutions. we are using findOneAndUpdate